File: hackhive-backend/src/main/java/com/hackhive/team/service/TeamService.java
package com.hackhive.team.service;

import com.hackhive.team.dto.request.CreateTeamRequest;
import com.hackhive.team.dto.request.UpdateTeamRequest;
import com.hackhive.team.dto.response.TeamMemberResponse;
import com.hackhive.team.dto.response.TeamResponse;

import java.util.List;

public interface TeamService {

    // Team management

    TeamResponse createTeam(
            CreateTeamRequest request
    );

    TeamResponse updateTeam(
            Long teamId,
            UpdateTeamRequest request
    );

    void deleteTeam(
            Long teamId
    );

    TeamResponse getTeamById(
            Long teamId
    );

    // Team discovery

    List<TeamResponse> getAllTeams();

    List<TeamResponse> searchTeamsByName(
            String name
    );

    List<TeamResponse> searchTeamsByCollege(
            String collegeName
    );

    List<TeamResponse> getTeamsByEvent(
            Long eventId
    );

    List<TeamResponse> getOpenTeams();

    // Current student's teams

    List<TeamResponse> getMyTeams();

    // Team members

    List<TeamMemberResponse> getTeamMembers(
            Long teamId
    );
}

File: hackhive-frontend/src/services/teamService.js
import api from "../api/axios";

async function request(promise) {
    const response = await promise;
    return response.data?.data ?? response.data;
}

export const teamService = {
    // Teams CRUD
    createTeam: (data) => request(api.post("/teams", data)),
    updateTeam: (teamId, data) => request(api.put(`/teams/${teamId}`, data)),
    deleteTeam: (teamId) => request(api.delete(`/teams/${teamId}`)),
    getTeamById: (teamId) => request(api.get(`/teams/${teamId}`)),
    getAllTeams: () => request(api.get("/teams")),
    getMyTeams: () => request(api.get("/teams/my-teams")),
    getOpenTeams: () => request(api.get("/teams/open")),
    getTeamsByEvent: (eventId) => request(api.get(`/teams/event/${eventId}`)),
    getTeamMembers: (teamId) => request(api.get(`/teams/${teamId}/members`)),
    leaveTeam: (teamId) => request(api.delete(`/teams/${teamId}/leave`)),
    removeMember: (teamId, studentProfileId) => request(api.delete(`/teams/${teamId}/members/${studentProfileId}`)),

    // Team Join Requests
    sendJoinRequest: (teamId) => request(api.post(`/team-join-requests/teams/${teamId}`)),
    getMyJoinRequests: () => request(api.get("/team-join-requests/my-requests")),
    cancelJoinRequest: (requestId) => request(api.delete(`/team-join-requests/${requestId}`)),
    getTeamJoinRequests: (teamId) => request(api.get(`/team-join-requests/teams/${teamId}`)),
    approveJoinRequest: (requestId) => request(api.put(`/team-join-requests/${requestId}/approve`)),
    rejectJoinRequest: (requestId) => request(api.put(`/team-join-requests/${requestId}/reject`)),
};
